micromanager_gui/_gui_objects/_group_preset_table_widget.py

```diff
@@ -0,0 +1,98 @@
+        self.setEditTriggers(QtW.QTableWidget.NoEditTriggers)
+        self.setColumnCount(2)
+        self.setHorizontalHeaderLabels(["Group", "Preset"])
+
+
+class MMGroupPresetTableWidget(QtW.QWidget):
+    def __init__(self):
+        super().__init__()
+
+        self._mmc = _core.get_core_singleton()
+        self._mmc.events.systemConfigurationLoaded.connect(self._populate_table)
+        self.table_wdg = MainTable()
+        self.table_wdg.show()
+        self.setLayout(QVBoxLayout())
+        self.setContentsMargins(0, 0, 0, 0)
+        self.layout().addWidget(self.table_wdg)
+
+    def _on_system_cfg_loaded(self):
+        self._populate_table()
+
+    def _reset_table(self):
+        self._disconnect_wdgs()
+        self.table_wdg.clearContents()
+        self.table_wdg.setRowCount(0)
+        # self.table_wdg.setHorizontalHeaderLabels(["Group", "Preset"])
+
+    def _disconnect_wdgs(self):
+        for r in range(self.table_wdg.rowCount()):
+            wdg = self.table_wdg.cellWidget(r, 1)
+            if isinstance(wdg, PresetsWidget):
+                wdg.disconnect()
+
+    def _populate_table(self):
+        self._reset_table()
+        if groups := self._mmc.getAvailableConfigGroups():
+            for row, group in enumerate(groups):
+                self.table_wdg.insertRow(row)
+                self.table_wdg.setItem(row, 0, QtW.QTableWidgetItem(str(group)))
+                wdg = self.create_group_widget(group)
+                self.table_wdg.setCellWidget(row, 1, wdg)
+                if isinstance(wdg, PresetsWidget):
+                    wdg = wdg._combo
+                elif isinstance(wdg, PropertyWidget):
+                    wdg = wdg._value_widget
+                if not self._mmc.getCurrentConfig(group):
+                    wdg.setStyleSheet("color: magenta;")
+
+    def _get_cfg_data(self, group: str, preset: str):
+        """
+        Return last device-property-value for the preset and the
+        total number of device-property-value included in the preset.
+        """
+
+        for dev_prop_val_count, key in enumerate(
+            self._mmc.getConfigData(group, preset)
+        ):
+            dev = key[0]
+            prop = key[1]
+            val = key[2]
+        return dev, prop, val, (dev_prop_val_count + 1)
+
+    def create_group_widget(self, group: str):
+        """Return a widget depending on presets and device-property"""
+
+        # get group presets
+        presets = list(self._mmc.getAvailableConfigs(group))
+
+        if not presets:
+            return
+
+        # use only the first preset since device
+        # and property are the same for the presets
+        device, property, _, dev_prop_val_count = self._get_cfg_data(group, presets[0])
+
+        if len(presets) > 1 or dev_prop_val_count > 1:
+            # return PresetsWidget(group, text_color="white")
+            return PresetsWidget(group)
+        else:
+            return PropertyWidget(device, property)
```

Comment:
I have the same concerns here as with the PresetWidget PR, only checking the first one isn't safe. Is there any reason not to just have this become:

```suggestion
        return PresetsWidget(group)
```

Reply:
@ianhi 
> I have the same concerns here as with the PresetWidget PR, only checking the first one isn't safe. Is there any reason not to just have this become: ...

When I made this I was thinking that each presets should have the exact same properties...However I think that here we need to distinguish between `PresetsWidget()` and `PropertyWidget()`. `PresetsWidget()` is when we have _more then one property_ in the preset. Instead, If we _only have one_ (e.g. if I just want to have a light control slider or just the combobox for the binning) we want to create a `PropertyWidget()`.

Reply:
I'm a little bit confused, In particular by your slider group:

```
# Group: _slider_test
# Preset: NewPreset
ConfigGroup,_slider_test,NewPreset,Camera,TestProperty1,0.0000
```

I don't see why you would make that into a config group. As I see it the purpose of a config group is to give convenient named settings of potentially continuous things (and of course to group them together). So a more natural group would be:
```
# Group: _slider_test
# Preset: NewPreset
ConfigGroup, continuous_property,Low,Camera,TestProperty1,0.0

ConfigGroup, continuous_property,Med,Camera,TestProperty1,0.3

ConfigGroup, continuous_property,High,Camera,TestProperty1,0..8
```


> If we only have one (e.g. if I just want to have a light control slider or just the combobox for the binning) we want to create a PropertyWidget().

I don't think that a slider really represents the concept of presets well, in this case you should instead use the property browser rathre than shoehorning this into the groups and presets

Reply:
It seems like what would maybe be a better solution is a new tab where you can select arbitrary properties to include. So that you have the power of the proprety browser but customized to only the properties that are relevant.

Reply:
> I don't see why you would make that into a config group

something like this  `ConfigGroup,_slider_test,NewPreset,Camera,TestProperty1,0.0000` in the cfg of micromanager is considers it as a `ConfigGroup`. So I think it makes sense to create a `QSlider` (PropertyWidget()) instead of a QCombobBox (`PresetsWidget()`).

> So a more natural group would be:
  ```
  # Group: _slider_test
  # Preset: NewPreset
  ConfigGroup, continuous_property,Low,Camera,TestProperty1,0.0

  ConfigGroup, continuous_property,Med,Camera,TestProperty1,0.3

  ConfigGroup, continuous_property,High,Camera,TestProperty1,0..8
  ```

I agree with you on this and since in this group there is more than one `preset`, each `preset` is considered/created as a `PresetsWidget()`.

@ianhi Don't you agree? maybe I'm missing something...